support/cas-server-support-ldap-monitor/src/main/java/org/apereo/cas/monitor/config/LdapMonitorConfiguration.java

```diff
@@ -36,16 +42,36 @@
     private CasConfigurationProperties casProperties;
 
     @Bean
+    public ListFactoryBean pooledLdapConnectionFactoryHealthIndicatorListFactoryBean() {
+        val list = new ListFactoryBean() {
+            @Override
+            protected void destroyInstance(final List list) {
+                list.forEach(connectionFactory -> {
+                    ((ConnectionFactory) connectionFactory).close();
+                });
+            }
+        };
+        list.setSourceList(new ArrayList());
+        return list;
+    }
+
+    @Bean
+    @SneakyThrows
+    @Autowired
+    @Qualifier("pooledLdapConnectionFactoryHealthIndicatorListFactoryBean")
     @ConditionalOnEnabledHealthIndicator("pooledLdapConnectionFactoryHealthIndicator")
-    public CompositeHealthContributor pooledLdapConnectionFactoryHealthIndicator() {
+    public CompositeHealthContributor pooledLdapConnectionFactoryHealthIndicator(
+            final ListFactoryBean pooledLdapConnectionFactoryHealthIndicatorListFactoryBean) {
```

Comment:
Should be:

```
 @Qualifier("pooledLdapConnectionFactoryHealthIndicatorListFactoryBean")
 final ListFactoryBean pooledLdapConnectionFactoryHealthIndicatorListFactoryBean
```

The parameter that is injected is qualified; not the bean itself.